File: src/main/java/com/negosud/api/service/CommandeClientService.java
package com.negosud.api.service;

import com.negosud.api.dto.request.CommandeClientRequestDto;
import com.negosud.api.dto.request.LigneCommandeRequestDto;
import com.negosud.api.dto.request.StatutCommandeDto;
import com.negosud.api.dto.response.CommandeClientResponseDto;
import com.negosud.api.dto.response.LigneCommandeClientResponseDto;

import java.util.List;

public interface CommandeClientService {

    List<CommandeClientResponseDto> findAll();

    CommandeClientResponseDto findById(Long id);

    List<LigneCommandeClientResponseDto> findLignesByCommande(Long id);

    CommandeClientResponseDto create(CommandeClientRequestDto dto);

    CommandeClientResponseDto update(Long id, CommandeClientRequestDto dto);

    CommandeClientResponseDto changerStatut(Long id, StatutCommandeDto dto);

    LigneCommandeClientResponseDto ajouterLigne(Long id, LigneCommandeRequestDto dto);

    void supprimerLigne(Long commandeId, Long ligneId);

    void delete(Long id);
}

File: front/src/services/commandeClientService.js
import api from './api';

const BASE = '/commandes-clients';

export const commandeClientService = {
  findAll: () => api.get(BASE),
  findById: (id) => api.get(`${BASE}/${id}`),
  findLignes: (id) => api.get(`${BASE}/${id}/lignes`),
  create: (data) => api.post(BASE, data),
  update: (id, data) => api.put(`${BASE}/${id}`, data),
  changerStatut: (id, statut) => api.patch(`${BASE}/${id}/statut`, { statut }),
  ajouterLigne: (id, data) => api.post(`${BASE}/${id}/lignes`, data),
  supprimerLigne: (id, ligneId) => api.delete(`${BASE}/${id}/lignes/${ligneId}`),
  delete: (id) => api.delete(`${BASE}/${id}`),
};
